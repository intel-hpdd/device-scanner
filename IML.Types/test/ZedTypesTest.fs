// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

module IML.DeviceScannerDaemon.ZedTypesTest

open Fable.Import.Jest
open Thoth.Json

open IML.Types.ZedTypes
open IML.CommonLibrary

open Matchers
open IML.Types.Fixtures
open Fable.PowerPack

test "decode / encode pools" <| fun () ->
  fixtures.pools
    |> Decode.decodeString Zed.decoder
    |> Result.unwrap
    |> Zed.encoder
    |> Encode.encode 2
    |> toMatchSnapshot
