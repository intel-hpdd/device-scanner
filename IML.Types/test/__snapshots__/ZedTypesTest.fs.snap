// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decode / encode pools 1`] = `
"{
  \\"14919184393193585238\\": {
    \\"name\\": \\"test\\",
    \\"guid\\": \\"14919184393193585238\\",
    \\"health\\": \\"ONLINE\\",
    \\"hostname\\": \\"localhost.localdomain\\",
    \\"hostid\\": null,
    \\"state\\": \\"ACTIVE\\",
    \\"readonly\\": false,
    \\"size\\": \\"83886080\\",
    \\"vdev\\": {
      \\"Root\\": {
        \\"children\\": [
          {
            \\"Mirror\\": {
              \\"children\\": [
                {
                  \\"Disk\\": {
                    \\"guid\\": \\"0xBE4606AF1C39DC3F\\",
                    \\"state\\": \\"ONLINE\\",
                    \\"path\\": \\"/dev/sdb1\\",
                    \\"dev_id\\": \\"ata-VBOX_HARDDISK_081118FC1221NCJ6G8G1-part1\\",
                    \\"phys_path\\": \\"pci-0000:00:0d.0-ata-2.0\\",
                    \\"whole_disk\\": true,
                    \\"is_log\\": null
                  }
                },
                {
                  \\"Disk\\": {
                    \\"guid\\": \\"0xCC43D91716DA2522\\",
                    \\"state\\": \\"ONLINE\\",
                    \\"path\\": \\"/dev/sdc1\\",
                    \\"dev_id\\": \\"ata-VBOX_HARDDISK_081118FC1221NCJ6G8G2-part1\\",
                    \\"phys_path\\": \\"pci-0000:00:0d.0-ata-3.0\\",
                    \\"whole_disk\\": true,
                    \\"is_log\\": null
                  }
                }
              ],
              \\"is_log\\": false
            }
          }
        ],
        \\"spares\\": [
          {
            \\"Disk\\": {
              \\"guid\\": \\"0x4DD5D18F6C1F6B34\\",
              \\"state\\": \\"ONLINE\\",
              \\"path\\": \\"/dev/sde1\\",
              \\"dev_id\\": \\"ata-VBOX_HARDDISK_081118FC1221NCJ6G8G4-part1\\",
              \\"phys_path\\": \\"pci-0000:00:0d.0-ata-5.0\\",
              \\"whole_disk\\": true,
              \\"is_log\\": null
            }
          },
          {
            \\"Disk\\": {
              \\"guid\\": \\"0x58650FD679A8842D\\",
              \\"state\\": \\"ONLINE\\",
              \\"path\\": \\"/dev/sdf1\\",
              \\"dev_id\\": \\"ata-VBOX_HARDDISK_081118FC1221NCJ6G8G5-part1\\",
              \\"phys_path\\": \\"pci-0000:00:0d.0-ata-6.0\\",
              \\"whole_disk\\": true,
              \\"is_log\\": null
            }
          }
        ],
        \\"cache\\": [
          {
            \\"Disk\\": {
              \\"guid\\": \\"0x70008C8B94F8AFA8\\",
              \\"state\\": \\"ONLINE\\",
              \\"path\\": \\"/dev/sdd1\\",
              \\"dev_id\\": \\"ata-VBOX_HARDDISK_081118FC1221NCJ6G8G3-part1\\",
              \\"phys_path\\": \\"pci-0000:00:0d.0-ata-4.0\\",
              \\"whole_disk\\": true,
              \\"is_log\\": null
            }
          }
        ]
      }
    },
    \\"props\\": [],
    \\"datasets\\": [
      {
        \\"name\\": \\"test/ds2\\",
        \\"guid\\": \\"11387853141151442667\\",
        \\"kind\\": \\"filesystem\\",
        \\"props\\": [
          {
            \\"name\\": \\"name\\",
            \\"value\\": \\"test/ds2\\"
          },
          {
            \\"name\\": \\"type\\",
            \\"value\\": \\"filesystem\\"
          },
          {
            \\"name\\": \\"creation\\",
            \\"value\\": \\"1522333734\\"
          },
          {
            \\"name\\": \\"used\\",
            \\"value\\": \\"24576\\"
          },
          {
            \\"name\\": \\"available\\",
            \\"value\\": \\"41751040\\"
          },
          {
            \\"name\\": \\"referenced\\",
            \\"value\\": \\"24576\\"
          },
          {
            \\"name\\": \\"compressratio\\",
            \\"value\\": \\"1.00x\\"
          },
          {
            \\"name\\": \\"mounted\\",
            \\"value\\": \\"yes\\"
          },
          {
            \\"name\\": \\"quota\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"reservation\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"recordsize\\",
            \\"value\\": \\"131072\\"
          },
          {
            \\"name\\": \\"mountpoint\\",
            \\"value\\": \\"/test/ds2\\"
          },
          {
            \\"name\\": \\"sharenfs\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"checksum\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"compression\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"atime\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"devices\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"exec\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"setuid\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"readonly\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"zoned\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"snapdir\\",
            \\"value\\": \\"hidden\\"
          },
          {
            \\"name\\": \\"aclinherit\\",
            \\"value\\": \\"restricted\\"
          },
          {
            \\"name\\": \\"createtxg\\",
            \\"value\\": \\"6638\\"
          },
          {
            \\"name\\": \\"canmount\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"xattr\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"copies\\",
            \\"value\\": \\"1\\"
          },
          {
            \\"name\\": \\"version\\",
            \\"value\\": \\"5\\"
          },
          {
            \\"name\\": \\"utf8only\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"normalization\\",
            \\"value\\": \\"none\\"
          },
          {
            \\"name\\": \\"casesensitivity\\",
            \\"value\\": \\"sensitive\\"
          },
          {
            \\"name\\": \\"vscan\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"nbmand\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"sharesmb\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"refquota\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"refreservation\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"guid\\",
            \\"value\\": \\"11387853141151442667\\"
          },
          {
            \\"name\\": \\"primarycache\\",
            \\"value\\": \\"all\\"
          },
          {
            \\"name\\": \\"secondarycache\\",
            \\"value\\": \\"all\\"
          },
          {
            \\"name\\": \\"usedbysnapshots\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"usedbydataset\\",
            \\"value\\": \\"24576\\"
          },
          {
            \\"name\\": \\"usedbychildren\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"usedbyrefreservation\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"logbias\\",
            \\"value\\": \\"latency\\"
          },
          {
            \\"name\\": \\"dedup\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"mlslabel\\",
            \\"value\\": \\"none\\"
          },
          {
            \\"name\\": \\"sync\\",
            \\"value\\": \\"standard\\"
          },
          {
            \\"name\\": \\"dnodesize\\",
            \\"value\\": \\"legacy\\"
          },
          {
            \\"name\\": \\"refcompressratio\\",
            \\"value\\": \\"1.00x\\"
          },
          {
            \\"name\\": \\"written\\",
            \\"value\\": \\"24576\\"
          },
          {
            \\"name\\": \\"logicalused\\",
            \\"value\\": \\"12288\\"
          },
          {
            \\"name\\": \\"logicalreferenced\\",
            \\"value\\": \\"12288\\"
          },
          {
            \\"name\\": \\"volmode\\",
            \\"value\\": \\"default\\"
          },
          {
            \\"name\\": \\"filesystem_limit\\",
            \\"value\\": \\"18446744073709551615\\"
          },
          {
            \\"name\\": \\"snapshot_limit\\",
            \\"value\\": \\"18446744073709551615\\"
          },
          {
            \\"name\\": \\"filesystem_count\\",
            \\"value\\": \\"18446744073709551615\\"
          },
          {
            \\"name\\": \\"snapshot_count\\",
            \\"value\\": \\"18446744073709551615\\"
          },
          {
            \\"name\\": \\"snapdev\\",
            \\"value\\": \\"hidden\\"
          },
          {
            \\"name\\": \\"acltype\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"context\\",
            \\"value\\": \\"none\\"
          },
          {
            \\"name\\": \\"fscontext\\",
            \\"value\\": \\"none\\"
          },
          {
            \\"name\\": \\"defcontext\\",
            \\"value\\": \\"none\\"
          },
          {
            \\"name\\": \\"rootcontext\\",
            \\"value\\": \\"none\\"
          },
          {
            \\"name\\": \\"relatime\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"redundant_metadata\\",
            \\"value\\": \\"all\\"
          },
          {
            \\"name\\": \\"overlay\\",
            \\"value\\": \\"off\\"
          }
        ]
      },
      {
        \\"name\\": \\"test/ds\\",
        \\"guid\\": \\"9760213047416233279\\",
        \\"kind\\": \\"filesystem\\",
        \\"props\\": [
          {
            \\"name\\": \\"name\\",
            \\"value\\": \\"test/ds\\"
          },
          {
            \\"name\\": \\"type\\",
            \\"value\\": \\"filesystem\\"
          },
          {
            \\"name\\": \\"creation\\",
            \\"value\\": \\"1521929884\\"
          },
          {
            \\"name\\": \\"used\\",
            \\"value\\": \\"24576\\"
          },
          {
            \\"name\\": \\"available\\",
            \\"value\\": \\"41751040\\"
          },
          {
            \\"name\\": \\"referenced\\",
            \\"value\\": \\"24576\\"
          },
          {
            \\"name\\": \\"compressratio\\",
            \\"value\\": \\"1.00x\\"
          },
          {
            \\"name\\": \\"mounted\\",
            \\"value\\": \\"yes\\"
          },
          {
            \\"name\\": \\"quota\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"reservation\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"recordsize\\",
            \\"value\\": \\"131072\\"
          },
          {
            \\"name\\": \\"mountpoint\\",
            \\"value\\": \\"/test/ds\\"
          },
          {
            \\"name\\": \\"sharenfs\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"checksum\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"compression\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"atime\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"devices\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"exec\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"setuid\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"readonly\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"zoned\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"snapdir\\",
            \\"value\\": \\"hidden\\"
          },
          {
            \\"name\\": \\"aclinherit\\",
            \\"value\\": \\"restricted\\"
          },
          {
            \\"name\\": \\"createtxg\\",
            \\"value\\": \\"6\\"
          },
          {
            \\"name\\": \\"canmount\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"xattr\\",
            \\"value\\": \\"on\\"
          },
          {
            \\"name\\": \\"copies\\",
            \\"value\\": \\"1\\"
          },
          {
            \\"name\\": \\"version\\",
            \\"value\\": \\"5\\"
          },
          {
            \\"name\\": \\"utf8only\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"normalization\\",
            \\"value\\": \\"none\\"
          },
          {
            \\"name\\": \\"casesensitivity\\",
            \\"value\\": \\"sensitive\\"
          },
          {
            \\"name\\": \\"vscan\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"nbmand\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"sharesmb\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"refquota\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"refreservation\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"guid\\",
            \\"value\\": \\"9760213047416233279\\"
          },
          {
            \\"name\\": \\"primarycache\\",
            \\"value\\": \\"all\\"
          },
          {
            \\"name\\": \\"secondarycache\\",
            \\"value\\": \\"all\\"
          },
          {
            \\"name\\": \\"usedbysnapshots\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"usedbydataset\\",
            \\"value\\": \\"24576\\"
          },
          {
            \\"name\\": \\"usedbychildren\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"usedbyrefreservation\\",
            \\"value\\": \\"0\\"
          },
          {
            \\"name\\": \\"logbias\\",
            \\"value\\": \\"latency\\"
          },
          {
            \\"name\\": \\"dedup\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"mlslabel\\",
            \\"value\\": \\"none\\"
          },
          {
            \\"name\\": \\"sync\\",
            \\"value\\": \\"standard\\"
          },
          {
            \\"name\\": \\"dnodesize\\",
            \\"value\\": \\"legacy\\"
          },
          {
            \\"name\\": \\"refcompressratio\\",
            \\"value\\": \\"1.00x\\"
          },
          {
            \\"name\\": \\"written\\",
            \\"value\\": \\"24576\\"
          },
          {
            \\"name\\": \\"logicalused\\",
            \\"value\\": \\"12288\\"
          },
          {
            \\"name\\": \\"logicalreferenced\\",
            \\"value\\": \\"12288\\"
          },
          {
            \\"name\\": \\"volmode\\",
            \\"value\\": \\"default\\"
          },
          {
            \\"name\\": \\"filesystem_limit\\",
            \\"value\\": \\"18446744073709551615\\"
          },
          {
            \\"name\\": \\"snapshot_limit\\",
            \\"value\\": \\"18446744073709551615\\"
          },
          {
            \\"name\\": \\"filesystem_count\\",
            \\"value\\": \\"18446744073709551615\\"
          },
          {
            \\"name\\": \\"snapshot_count\\",
            \\"value\\": \\"18446744073709551615\\"
          },
          {
            \\"name\\": \\"snapdev\\",
            \\"value\\": \\"hidden\\"
          },
          {
            \\"name\\": \\"acltype\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"context\\",
            \\"value\\": \\"none\\"
          },
          {
            \\"name\\": \\"fscontext\\",
            \\"value\\": \\"none\\"
          },
          {
            \\"name\\": \\"defcontext\\",
            \\"value\\": \\"none\\"
          },
          {
            \\"name\\": \\"rootcontext\\",
            \\"value\\": \\"none\\"
          },
          {
            \\"name\\": \\"relatime\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"redundant_metadata\\",
            \\"value\\": \\"all\\"
          },
          {
            \\"name\\": \\"overlay\\",
            \\"value\\": \\"off\\"
          },
          {
            \\"name\\": \\"lustre:mgsnode\\",
            \\"value\\": \\"10.14.82.0@tcp:10.14.82.1@tcp\\"
          }
        ]
      }
    ]
  }
}"
`;
